### ISSUE TITLE
angular 2.0 tutorial impossible on windows 10

### ISSUE THREAD

**Issue by pwnball** [Urigo/angular2-meteor#301]
I just started a new project and wanted to give angular2 a swing with meteor. Unfortunately my enthusiasm was swiftly met with disappointment when I learned that even the first angular 2 step in the tutorial stopped me in my tracks.
https://www.angular-meteor.com/tutorials/socially/angular2/bootstrapping

The command : 
`meteor npm install --save angular2-meteor @angular/platform-browser-dynamic`
gives back this result:

> npm ERR! Windows_NT 10.0.10586
> npm ERR! argv "C:\\Users\\Daan\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\bin\\\\node.exe" "C:\\Users\\Daan\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.3.3\\mt-os.windows.x86_32\\dev_bundle\\lib\\node_modules\\n
> pm\\bin\\npm-cli.js" "install" "--save" "angular2-meteor" "@angular/platform-browser-dynamic"
> npm ERR! node v0.10.45
> npm ERR! npm  v2.15.1
> npm ERR! code EPEERINVALID
> 
> npm ERR! peerinvalid The package @angular/core@2.0.0-rc.2 does not satisfy its siblings' peerDependencies requirements!
> npm ERR! peerinvalid Peer @angular/common@2.0.0-rc.2 wants @angular/core@2.0.0-rc.2
> 

This is a fresh project.
I am using windows 10 with meteor 1.3.2.4. I would use 1.3.3, but that version does not seem to work with windows 7 and 10 in its current state.

Thanks! :)

**Comment by nanomoffet** [comment]
RC.2 seemed to break a lot. I couldn't start a new project on OS X either until I reverted all the Angular dependencies in my package.json to rc.1

**Comment by pwnball** [comment]
@nanomoffet I have no clue how to revert anything, I just started on angular2-meteor.. but I havent gotten any project to actually work.. they all give peerinvalid errors :(

**Comment by nanomoffet** [comment]
There was a comment on another issue about that being fixed, though I don't know when it'll be released. Here's a copy of my package.json that uses rc.1. If you start a fresh project up to just before it says to use npm install, then replace your package.json contents with what's below, do npm install, you should be able to continue on from there. At least, it worked for me yesterday.

***

```
package.json
{
  "name": "socially",
  "private": true,
  "scripts": {
    "start": "meteor run"
  },
  "dependencies": {
    "@angular/common": "2.0.0-rc.1",
    "@angular/compiler": "2.0.0-rc.1",
    "@angular/core": "2.0.0-rc.1",
    "@angular/platform-browser": "2.0.0-rc.1",
    "@angular/platform-browser-dynamic": "2.0.0-rc.1",
    "@angular/router":"3.0.0-alpha.3",
    "angular2-meteor": "0.5.5",
    "angular2-meteor-auto-bootstrap": "0.5.5",
    "angular2-meteor-polyfills": "^0.1.1",
    "es6-shim": "0.35.1",
    "meteor-node-stubs": "^0.2.3",
    "reflect-metadata": "0.1.2",
    "rxjs": "5.0.0-beta.6",
    "typings": "^1.3.0",
    "zone.js": "0.6.12"
  }
}
```

**Comment by pwnball** [comment]
@nanomoffet thanks for your package.json. This did in fact allow me to succesfully npm install ngular2-meteor and @angular/platform-browser-dynamic.

Unfortunately I am still stuck on the bootstrapping part of the tutorial.

I have setup app.ts, app.html and index.html all according to the tutorial. I then run meteor as the tutorial suggests to test out if angular 2 is now succesfully doing its thing..

Unfortunately it is not and I am met with this error:
EXCEPTION: No Directive annotation found on Socially

After some googling I would think that in the current Socially tutorial this should no longer be a problem. Really hyped up to get started on angular 2, its a darn shame its this hard to even get started :)

**Comment by nanomoffet** [comment]
@ panball Can you post your app.ts?

**Comment by pwnball** [comment]
@nanomoffet  certainly!
```

import 'reflect-metadata';
import { Component } from '@angular/core';
import { bootstrap } from '@angular/platform-browser-dynamic';

@Component({
    selector: 'app',
    templateUrl: '/client/app.html'
})
class Socially { }

bootstrap(Socially);
```

**Comment by Karnich** [comment]
Ive experienced various of errors when trying to start the tutorial at step 0 and downloading the step 18 code.  Finally gotten it working. Seems like the newest meteor have a problem. I install all missing dependencies. and in the .meteor folder i changed the text in the release file from METEOR@1.3.3.1 -> METEOR@1.3.2.4 
While looking for the errors i upgraded both npm and node to the newest versions. And im also on windows 10.

**Comment by pwnball** [comment]
@Karnich nice that it works for you :)
I am still struggling on the same issue, I create my project as 1.3.2.4 from the get-go as 1.3.3 doesnt seem to work well on windows at the moment. My npm and node are at the latest version and I used the package json supplied in this threat.

**Comment by Karnich** [comment]
@pwnball i must admin i spoke too soon. I Got it building after downgrading meteor. But getting the "more than one platform" error. I can passe html but my constructor in my AppComponent (socially) does not run. In general i Think this is not useable right now. The router is 2 versions behind and every dependeny needs to be upgraded to rc2. Angular2 have had a shaky api for a while and i Think right now im Gonna wait for it too become more stable and these packages get up to date. Spend the whole Night yesterday trying to make stuff work without any luck

**Comment by NOaMTL** [comment]
This is how I solved the issue :
Let's assume this is a fresh project. Follow the tutorial. After running these commands :
`meteor remove blaze-html-templates`
`meteor add angular2-compilers barbatus:angular2-runtime`

Instead of :

`meteor npm install --save angular2-meteor @angular/platform-browser-dynamic`

use :

`meteor npm install --save angular2-meteor @angular/platform-browser-dynamic@2.0.0-rc.1`

Then use the :

`meteor npm install`

You can try now to launch your app using "meteor" command, if in your browser console you see : 

Uncaught Error: Cannot find module 'angular2-meteor-polyfills/browser
Then run this command (you dont need to import this in your app) :

`meteor npm install --save angular2-meteor-polyfills`

Now continue the tutorial. When you readh the 5th step (routing), instead of "meteor npm install --save @angular/router-deprecated" use : 

`meteor npm install --save @angular/router@3.0.0-alpha.3`

Now the fun part... 
This is how i manage my routes with this library :
Let's say you have a router main file (it is like you app router file, all the routes will be "injected" inside) :
```
import { provideRouter, RouterConfig } from '@angular/router';
import { HomeRoutes } from './imports/home/home.routes';
import { TestRoutes } from './imports/test/test.routes';


export const routes: RouterConfig = [
  ...HomeRoutes,
  ...TestRoutes
];
export const APP_ROUTER_PROVIDERS = [
  provideRouter(routes)
];
```

The "..." are important, they mean merging in ES6 syntax.
Now this is my test.routes.ts file :

```
import { RouterConfig } from '@angular/router';
import { TestComponent } from './test.component';

export const TestRoutes: RouterConfig = [
  { path: '/test', component: TestComponent }

];
```

And this is my app.ts file :

```
import 'reflect-metadata';
import { Component, provide } from '@angular/core';
import { APP_BASE_HREF } from '@angular/common';
import { bootstrap } from 'angular2-meteor-auto-bootstrap';
import { chatMessage }   from '../collections/chatmessage';
import { Mongo }     from 'meteor/mongo';
import { ROUTER_DIRECTIVES, provideRouter, RouterConfig } from '@angular/router';
import { APP_ROUTER_PROVIDERS } from './routes';
 
@Component({
  selector: 'app',
  templateUrl: '/client/app.html',
  directives: [ROUTER_DIRECTIVES]
})
class Bravery { }
 
bootstrap(Bravery, [APP_ROUTER_PROVIDERS, provide(APP_BASE_HREF, { useValue: '/' })]);
```

You can find more infos about the router : [here](https://angular.io/docs/ts/latest/guide/router.html)
(I don't know if you will find more issues later in the tutorial)

**Comment by nanomoffet** [comment]
I'm curious how you set up the new router to pull the parameter from the
URL for the party detail view. I fumbled my way through similar steps and
got the new router working except for passing the parameters in the
controller to get a single entry since the ng2 data services are set up
very different from the tutorial.

Note that the step on user Auth imports some other stuff that has a listed
dependency on router-deprecated.

**Comment by NOaMTL** [comment]
I will try later today to do it, there is a good example from the official doc on how to get the parameters : [https://angular.io/docs/ts/latest/guide/router.html#!#query-parameter](https://angular.io/docs/ts/latest/guide/router.html#!#query-parameter)

**Comment by NOaMTL** [comment]
ok so this is my code for retrieving a parameter  (assuming my route is like "/test/:id"):

```
import 'reflect-metadata';
import { Component, OnInit, OnDestroy } from '@angular/core';
import { Footer } from '../footer/footer.ts';
import { MainMenu } from '../menus/main-menu.ts';
import { ROUTER_DIRECTIVES } from '@angular/router';
import { Router, ActivatedRoute }       from '@angular/router';

@Component({
  selector: 'test-detail',
  templateUrl: '/client/imports/test/test-detail.html',
  directives: [Footer, MainMenu, ROUTER_DIRECTIVES]
})



    export class TestDetailComponent implements OnInit, OnDestroy  {

  private sub: any;

  constructor(
    private route: ActivatedRoute,
    private router: Router) {}

  ngOnInit() {
    this.sub = this.route.params.subscribe(params => {
      this.selectedId = +params['id']; // use {{selectedId}} on your view to display it
     });
  }

  ngOnDestroy() {
  }

}
```

**Comment by acefed** [comment]
@NOaMTL Thanks for getting me on track again with the installation of the tutorial o/

**Comment by nanomoffet** [comment]
Thanks for the code! I'm curious, what is routerState doing?

**Comment by nickprusov** [comment]
can someone post *.ts configuration files for tutorial steps 5.1 - 5.16 after importing meteor npm install --save @angular/router@3.0.0-alpha.3

**Comment by pwnball** [comment]
@nickprusov I would like that as well... Im toying around with it as we speak, but theres a few things that manage to dodge my grasp. Not sure if I will or will not get this working, but having a few files as a reference would greatly speed up the process without a doubt :)

**Comment by pwnball** [comment]
@NOaMTL 
You have added `import { TestComponent } from './test.component';` to your test.routes.ts file. You don't describe this file at all though.. am I missing something crucial here?

**Comment by NOaMTL** [comment]
@pwnball This `import { TestComponent } from './test.component';` is just a custom module that i made for testing angular and meteor features, it has no relation with the tutorial. In my exemple above, i am using the tutorial for the basics but I was trying to make a custom app. Also I am not organizing the files exactly like in the tutotial, the important thing is to understand how the router is working (if you plan to use the new one, and not the deprectated one). So for more clarity i put a router file in the folder of ech module (so for the tutorial it could be a parties-list.router.ts in the parties-list folder. And then in the main router file, you inject all the  routes using the "export" class name ("TestRoutes" in my exemple).
@nanomoffet @pwnball If you are still stuck at step5 I will try to find time and do the tutorial with the new router and post all the files

**Comment by pbastowski** [comment]
@pwnball Dan, the routing config in ng2 router v3 alpha sucks. This feels like going backwards to Angular 1's ng-route, with all the routes having to be declared in one file. Except that you have to declare them all in separate files, next to each of your components, and then manually combine them in the top level component, just so you can pass them export them again only to be imported in a the top-level main.ts where they are given as an argument to the provideRouter function that you in turn have to list in the providers array of the bootstrap function.

Yeah, this is a whinge :)

I got it working with Meteor, finally, but this is not cool!

Although it is possible to use provideRouter in a lower level component, it would seem that a catch-all path like "**" does not seem to work if specified anywhere other than during bootstrap. Perhaps this is something that I'm doing wrong or a bug in the Alpha release of the v3 router, but still, it is not obvious and I could not find anything about this in the "latest" [docs](https://angular.io/docs/ts/latest/guide/router.html#!#index).

Anyone else found a nicer way to define routes? Please share here. Thanks.

**Comment by pbastowski** [comment]
@pwnball Check out https://github.com/ngrx/router.

**Comment by debovema** [comment]
@Urigo : is there any plan to update the tutorial to use a non-deprecated router and to make the tutorial working with latest meteor version (1.3.4) ? Thanks in advance.

**Comment by JohnBanfield** [comment]
Hi I'm giving up on the angular 2 tutorial until it works.  I'll try the angular 1 one.

**Comment by mackbrowne** [comment]
Sadly I'm in the same situation.  Spent almost a day going through github issues and trying to get my project just to launch without error.  Let me know when the examples are working properly and i'd love to try again.

**Comment by barbatus** [comment]
Should be fixed  in 0.6.0.

**Comment by mrp100461** [comment]
its not
